epix/run_epix.py

```diff
@@ -19,14 +19,28 @@ def main(args, return_df=False, return_wfsim_instructions=False, strax=False):
         tnow = starttime
 
     # Loading data:
-    inter, n_simulated_events = epix.loader(args['path'],
-                                            args['file_name'],
-                                            args['debug'],
-                                            outer_cylinder=args['outer_cylinder'],
-                                            kwargs_uproot_arrays={'entry_start': args['entry_start'],
-                                                                  'entry_stop': args['entry_stop']},
-                                            cut_by_eventid=args['cut_by_eventid']
-                                            )
+
+    if args["file_name"].endswith(".root"):
+        #the usual case: root files:
+        inter, n_simulated_events = epix.loader(args['path'],
+                                                args['file_name'],
+                                                args['debug'],
+                                                outer_cylinder=args['outer_cylinder'],
+                                                kwargs_uproot_arrays={'entry_start': args['entry_start'],
+                                                                    'entry_stop': args['entry_stop']},
+                                                cut_by_eventid=args['cut_by_eventid']
+                                                )
+    elif args["file_name"].endswith(".csv"):
+        #the case of manualy crafted instructions 
+        print("Load instructions from a csv file!")
+        inter, n_simulated_events = epix.csv_loader(args['path'],
+                                                    args['file_name'],
+                                                    args['debug'],
+                                                    outer_cylinder=args['outer_cylinder'],
+                                                    kwargs={'entry_start': args['entry_start'],
+                                                            'entry_stop': args['entry_stop']},
+                                                    cut_by_eventid=args['cut_by_eventid'])
```

Comment:
Could we change this a bit I think the logic should be always from being very general to becoming more and more specific. In this case this would mean you have a single loader function called in run_epix. And this single loader function then branches out to load either csv or root files depending on the users input. 
In this way you keep the main program short and only if a developer is interested in the loading process itself he/she can have a look what the "loader" is actually doing.

Reply:
Hence in this case it would make sense to add a new loader class which first checks some general stuff e.g. is it a valid path etc before calling the correct load function depending on the user input.

Reply:
Good idea! I modified it in this commit: https://github.com/XENONnT/epix/pull/42/commits/e8fb8412ead766e146135b73c18b2b0c30fc36f3 .  At the moment only a basic test if the file exists is performed. The logic in run_epix.py should be clean again.